### PR TITLE
Add keyboard cover60

### PR DESCRIPTION
**Pull request by TW59420**
## Description
MWStudio is a custom keyboard enthusiast
Add keyboard cover60 PCB

## Types of Changes
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mwstudio/cover60/rules.mk
Comment (zvecr) [inline keyboards/mwstudio/cover60/rules.mk:1]: Should match <https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/rules.mk>

File: keyboards/mwstudio/cover60/halconf.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define HAL_USE_PWM TRUE
+// #define HAL_USE_PAL TRUE
+// #define HAL_USE_GPT TRUE
+// #define HAL_USE_I2C TRUE
```
Comment (zvecr) [inline keyboards/mwstudio/cover60/halconf.h:22]:
```suggestion
```

File: keyboards/mwstudio/cover60/config.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#define FORCE_NKRO
+
+#define RGB_DI_PIN B8
+#define DRIVER_LED_TOTAL 19
```
Comment (zvecr) [inline keyboards/mwstudio/cover60/config.h:21]: This setting isnt a thing
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by TW59420** [comment]
@drashna Hi, Can you help me merge branches

**Review by drashna — APPROVED** [review]

File: keyboards/mwstudio/cover60/config.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#define FORCE_NKRO
+
+#define WS2812_DI_PIN B8
+#define RGBLED_NUM 19
```
Comment (fauxpark) [inline keyboards/mwstudio/cover60/config.h:21]: These can be moved to info.json.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mwstudio/cover60/info.json [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    },
+    "matrix_pins": {
+        "cols": ["B6", "A10", "A9", "A8", "B15", "B14", "B13", "A4", "A5", "A6", "A7", "B10", "B11", "A3" ],
+        "rows": ["A2", "B12", "B5", "B7", "B9" ]
+    },
+    "processor": "STM32F103",
+    "bootloader": "stm32duino",
+    "force_nkro": true,
+    "ws2812_di_pin": "B8",
+    "rgbled_num": 19,
```
Comment (fauxpark) [inline keyboards/mwstudio/cover60/info.json:23]:
```suggestion
```
These are not valid.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mwstudio/cover60/config.h (hunk 3 above)
Comment (TW59420) [reply keyboards/mwstudio/cover60/config.h]: I moved to info.json,But the compilation does not pass. [QMK Firmware](https://docs.qmk.fm/#/)not updated

Comment (fauxpark) [reply keyboards/mwstudio/cover60/config.h]: `FORCE_NKRO` -> `usb.force_nkro`
`WS2812_DI_PIN` -> `ws2812.pin`
`RGBLED_NUM` -> `rgblight.led_count`

File: keyboards/mwstudio/cover60/info.json [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    "bootloader": "stm32duino",
+    "usb.force_nkro": true,
+    "ws2812.pin": "B8",
+    "rgblight.led_count": 19,
+    "url": "",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x6001",
+        "vid": "0x7BA1"
+    },
```
Comment (fauxpark) [inline keyboards/mwstudio/cover60/info.json:29]:
```suggestion
    "ws2812": {
        "pin": "B8"
    },
    "rgblight": {
        "led_count": 19,
    },
    "url": "",
    "usb": {
        "device_version": "1.0.0",
        "pid": "0x6001",
        "vid": "0x7BA1",
        "force_nkro": true
    },
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mwstudio/cover60/info.json (hunk 5 above)
Comment (TW59420) [reply keyboards/mwstudio/cover60/info.json]: thank you, i get

File: keyboards/mwstudio/cover60/readme.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Cover60
+
+![Cover60](https://imgur.com/4FOL8zj)
```
Comment (fauxpark) [inline keyboards/mwstudio/cover60/readme.md:3]:
```suggestion
![Cover60](https://i.imgur.com/4FOL8zj.png)
```
Needs to be a direct link.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mwstudio/cover60/readme.md (hunk 6 above)
Comment (TW59420) [reply keyboards/mwstudio/cover60/readme.md:3]: thanks

File: keyboards/mwstudio/cover60/readme.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
+# Cover60
+
+![Cover60](https://i.imgur.com/4FOL8zj.png)
```
Comment (dunk2k) [inline keyboards/mwstudio/cover60/readme.md:3]:
```suggestion
![Cover60](https://i.imgur.com/4FOL8zjh.png)
```

File: keyboards/mwstudio/cover60/readme.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+    make mwstudio/cover60:default 
+
+Flashing example for this keyboard:
+
+    make mwstudio/cover60:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
```
Comment (dunk2k) [inline keyboards/mwstudio/cover60/readme.md:20]: bootmagic is enabled in `info.json`
```suggestion
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/mwstudio/cover60/info.json [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+        "cols": ["B6", "A10", "A9", "A8", "B15", "B14", "B13", "A4", "A5", "A6", "A7", "B10", "B11", "A3" ],
+        "rows": ["A2", "B12", "B5", "B7", "B9" ]
+    },
+    "processor": "STM32F103",
+    "bootloader": "stm32duino",
+    "ws2812": {
+        "pin": "B8"
+    },
+    "rgblight": {
+        "led_count": 19,
```
Comment (dunk2k) [inline keyboards/mwstudio/cover60/info.json:25]:
```suggestion
        "led_count": 19
```

File: keyboards/mwstudio/cover60/info.json [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+    },
+    "url": "",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x6001",
+        "vid": "0x7BA1",
+        "force_nkro": true
+    },
+    "layouts": {
```
Comment (dunk2k) [inline keyboards/mwstudio/cover60/info.json:34]: Looking at the PCB image provided in `readme.md`, this board has more than one key arrangement available. Would you be able to add additional key arrangements to `layouts`?

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/mwstudio/cover60/info.json [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+        "LAYOUT": {
+            "layout": [
+				{ "matrix": [0, 0], "x": 0, "y": 1 },
```
Comment (drashna) [inline keyboards/mwstudio/cover60/info.json:37]:
```suggestion
                { "matrix": [0, 0], "x": 0, "y": 1 },
```

File: keyboards/mwstudio/cover60/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT(
```
Comment (drashna) [inline keyboards/mwstudio/cover60/keymaps/default/keymap.c:20]:
```suggestion
    [0] = LAYOUT(
```

File: keyboards/mwstudio/cover60/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,     KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,     KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,     KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,     KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_UP,   KC_RSFT,
+        KC_LCTL, KC_LGUI, KC_LALT,                   KC_SPACE,KC_SPACE,          KC_SPACE,KC_RALT, MO(1),   KC_LEFT, KC_DOWN, KC_RIGHT
+    ),
+
+	[1] = LAYOUT(
```
Comment (drashna) [inline keyboards/mwstudio/cover60/keymaps/default/keymap.c:28]:
```suggestion
    [1] = LAYOUT(
```

File: keyboards/mwstudio/cover60/keymaps/via/keymap.c [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT(
```
Comment (drashna) [inline keyboards/mwstudio/cover60/keymaps/via/keymap.c:21]:
```suggestion
    [0] = LAYOUT(
```

File: keyboards/mwstudio/cover60/keymaps/via/keymap.c [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,     KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,     KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,     KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,     KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_UP,   KC_RSFT,
+        KC_LCTL, KC_LGUI, KC_LALT,                   KC_SPACE,KC_SPACE,          KC_SPACE,KC_RALT, MO(1),   KC_LEFT, KC_DOWN, KC_RIGHT
+    ),
+
+	[1] = LAYOUT(
```
Comment (drashna) [inline keyboards/mwstudio/cover60/keymaps/via/keymap.c:29]:
```suggestion
    [1] = LAYOUT(
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mwstudio/cover60/info.json [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+    },
+    "url": "",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x6001",
+        "vid": "0x7BA1",
+        "force_nkro": true
+    },
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/mwstudio/cover60/info.json:35]:
```suggestion
        "LAYOUT_all": {
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Pull request closed without merging**